pilot/pkg/security/model/authentication.go

```diff
@@ -268,7 +270,8 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, proxy *model.Prox
 	// TODO: if user explicitly specifies SANs - should we alter his explicit config by adding all spifee aliases?
 	matchSAN := util.StringToExactMatch(subjectAltNames)
 	if len(trustDomainAliases) > 0 {
-		matchSAN = append(matchSAN, util.StringToPrefixMatch(appendURIPrefixToTrustDomain(trustDomainAliases))...)
+		matchSAN = append(matchSAN, util.StringToPrefixMatch(
+			appendURIPrefixToTrustDomain(dedupTrustDomains(trustDomainAliases)))...)
```

Comment:
I'm thinking shall we do it even earlier? Can you look a bit upper of the call chain of `ApplyToCommonTLSContext` see if there's an appropriate place?

That way, other future callers does not need to memorize calling (or implmementing) this dedup function.

Reply:
Thanks for the comment! Moved to where John suggests https://github.com/istio/istio/pull/32265#pullrequestreview-639223896. Please kindly see the details below.